### PR TITLE
Created Sign in screen GUI

### PR DESCRIPTION
**Pull request by cristiangligan**

### CODE REVIEW

File: src/view/SignInScreen.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+import javax.swing.*;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import java.awt.*;
+import java.awt.font.TextAttribute;
+import java.util.Map;
+
+public class SignInScreen extends JFrame {
+    private JPanel pnlMain = new JPanel();
```
Comment (SkillsDelivery) [inline src/view/SignInScreen.java:13]: You could safely make all data members final.

File: src/view/SignInScreen.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        springLayout.putConstraint(SpringLayout.WEST, lblAlreadyHaveAcc, 0, SpringLayout.WEST, lblProfilePic);
+        springLayout.putConstraint(SpringLayout.SOUTH, lblAlreadyHaveAcc, 0, SpringLayout.SOUTH, btnSignIn);
+
+        btnSignUp.setText("Sign up");
+        pnlMain.add(btnSignUp);
+        btnSignUp.setBorder(BorderFactory.createEmptyBorder());
+        btnSignUp.setRolloverEnabled(true);
+        btnSignUp.getModel().addChangeListener(e -> onHoverSignUp());
+        springLayout.putConstraint(SpringLayout.NORTH, btnSignUp, 0, SpringLayout.NORTH, lblAlreadyHaveAcc);
+        springLayout.putConstraint(SpringLayout.WEST, btnSignUp, 5, SpringLayout.EAST, lblAlreadyHaveAcc);
```
Comment (SkillsDelivery) [inline src/view/SignInScreen.java:67]: Magic number 5. Looks like this is a specific padding / spacing. Use a constant and name it specifically like SIGN_UP_BTN_SPACE

File: src/view/SignInScreen.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        springLayout.putConstraint(SpringLayout.SOUTH, btnSignUp, 0, SpringLayout.SOUTH, lblAlreadyHaveAcc);
+
+        this.pack();
+        this.setVisible(true);
+        this.setSize(new Dimension(460, 220));
```
Comment (SkillsDelivery) [inline src/view/SignInScreen.java:72]: Magic values for size. Create constants. Name then properly.

File: src/view/SignInScreen.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import java.awt.font.TextAttribute;
+import java.util.Map;
+
+public class SignInScreen extends JFrame {
+    private JPanel pnlMain = new JPanel();
+    private JLabel lblProfilePic = new JLabel();
+    private JTextField txtUsername = new JTextField();
+    private JTextField txtPassword = new JTextField();
+    private JButton btnSignIn = new JButton();
+    private JButton btnSignUp = new JButton();
```
[the remark this answers is not in the document]
Comment (cristiangligan) [reply src/view/SignInScreen.java:18]: Done

File: src/view/SignInScreen.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
+package view;
+
+import controller.Controller;
+
+import javax.swing.*;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
```
[the remark this answers is not in the document]
Comment (cristiangligan) [reply src/view/SignInScreen.java:7]: Done

File: src/view/SignInScreen.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
+package view;
+
+import controller.Controller;
+
+import javax.swing.*;
+import javax.swing.event.ChangeEvent;
```
[the remark this answers is not in the document]
Comment (cristiangligan) [reply src/view/SignInScreen.java:6]: Done

File: src/view/SignInScreen.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+    public SignInScreen(Controller controller) {
+        this.controller = controller;
+        setTitle("ChatApp");
+        SpringLayout springLayout = new SpringLayout();
+        setContentPane(pnlMain);
+        pnlMain.setLayout(springLayout);
+
+        Image image = new ImageIcon("src/images/blank-profile-picture.png").getImage();
+        Image newImage = image.getScaledInstance(100, 100, Image.SCALE_SMOOTH);
```
[the remark this answers is not in the document]
Comment (cristiangligan) [reply src/view/SignInScreen.java:30]: Done

File: src/view/SignInScreen.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        Icon icon = new ImageIcon(newImage);
+        lblProfilePic.setIcon(icon);
+        lblProfilePic.setBorder(BorderFactory.createLineBorder(Color.GRAY));
+        pnlMain.add(lblProfilePic);
+        springLayout.putConstraint(SpringLayout.NORTH, lblProfilePic, 20, SpringLayout.NORTH, pnlMain);
```
[the remark this answers is not in the document]
Comment (cristiangligan) [reply src/view/SignInScreen.java:35]: Done

File: src/view/SignInScreen.java (hunk 1 above)
Comment (cristiangligan) [reply src/view/SignInScreen.java:13]: Done

File: src/view/SignInScreen.java (hunk 2 above)
Comment (cristiangligan) [reply src/view/SignInScreen.java:67]: Done

File: src/view/SignInScreen.java (hunk 3 above)
Comment (cristiangligan) [reply src/view/SignInScreen.java:72]: Done

File: src/view/SignInScreen.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    private Controller controller;
+
+    public SignInScreen(Controller controller) {
+        this.controller = controller;
+        setTitle("ChatApp");
+        SpringLayout springLayout = new SpringLayout();
+        setContentPane(pnlMain);
+        pnlMain.setLayout(springLayout);
+
+        Image image = new ImageIcon("src/images/blank-profile-picture.png").getImage();
```
[the remark this answers is not in the document]
Comment (cristiangligan) [reply src/view/SignInScreen.java:29]: Done

**Pull request closed without merging**